### ISSUE TITLE
No hard coded paths,  please

### ISSUE THREAD

**Issue by mawg** [MahdiSafsafi/zcontrols#18]
Not everyone installed Delphi to the default path. Which leads to ...

In InspDemo.vrc

AmethystKamri VCLSTYLE "e:\\coding\\XE8\\Redist\\styles\\vcl\\Studio\\15.0\\Styles\\AmethystKamri.vsf"
CyanDusk VCLSTYLE "e:\\coding\\XE8\\Redist\\styles\\vcl\\Studio\\15.0\\Styles\\CyanDusk.vsf"
Luna VCLSTYLE "e:\\coding\\XE8\\Redist\\styles\\vcl\\Studio\\15.0\\Styles\\Luna.vsf"

[BRCC32 Error] InspDemo.vrc(66): file not found: C:\Users\Public\Documents\Embarcadero\Studio\15.0\Styles\AmethystKamri.vsf
[BRCC32 Error] InspDemo.vrc(67): file not found: C:\Users\Public\Documents\Embarcadero\Studio\15.0\Styles\CyanDusk.vsf
[BRCC32 Error] InspDemo.vrc(68): file not found: C:\Users\Public\Documents\Embarcadero\Studio\15.0\Styles\Luna.vsf

**Comment by pyscripter** [comment]
The vrc file is created by the compiler.  If you see the source code of 
zcontrols/Demo/InspDemo.dproj  it does not contain hard coded paths.  However it does specify the use of AmethystKamri and the other styles which apparently are not available in your Delphi installation or at least they are not present in the location expected by the compiler.  Just go to the Project options, Appearance, remove these styles and you should be OK.

**Comment by mawg** [comment]
Yes, I can see that the file is auto-generated.  When I editted it to my path, the file was overwritten when I built.  What I did was to copy those three files to the default location & everything was fine. is it worth making a small note in How To Install.txt?

--------

A separate question.:I have long been looking for something like the Object Inspector which I can use at run-time  to examine & change  (some of) the properties of my own objects, which are not all VCL controls. Can I do that with this component?

**Comment by pyscripter** [comment]
The inspector can be used to inspect and change any object.  The property Component is of type TObject and can be set to any Delphi object.

**Comment by MahdiSafsafi** [comment]
Hi there, 

As pyscripter pointed you out, there is no hard paths from the library. Only the compiler that made such paths. 
Please, select ```Project => Options => Appearance``` and then press OK. This will fix all paths.

Note: There is no need to remove styles. The demo enumerates the available styles and it doesn't force 
for a particular style.

For your second question, I believe that pyscripter explained it very well :)

**Comment by mawg** [comment]
Can you show a very basic demo of adding a non-VCL control class and showing its properties in the inspector?

**Comment by mawg** [comment]
I tried
  TMyClass = Class(TObject)
     public
       val :Integer;
       motto : String;
  End;
  myObject : TMyClass;

and

procedure TMain.FormCreate(Sender: TObject);
begin
  GetObjsList;
  EnumStyles;

  myObject := TMyClass.Create();
  myObject.val := 13;
  myObject.motto := 'Hello, world';

  zObjectInspector1.Component := TComponent(myObject);
end;

but my object was not shown in the object inspector :-(

**Comment by MahdiSafsafi** [comment]
That's because ```var``` is not a published property. Inspector works with properties and not variables/constants.

**Comment by mawg** [comment]
You are correct !!!  I changed it to

  TMyClass = Class(TObject)
     private
         myValue : Integer;
         myMotto : String;

     Published
        property  val :Integer read myValue write myValue;
        property  motto : String read myMotto write myMotto;
  End;

and now those properties of my object show in the Object Inspector ! 

Wow! I have been looking for something like this for over ten years; thank you *so* much !

**Comment by pyscripter** [comment]
Please close the issue.
